src/smk/viewer-leaflet/layer/layer-esri-feature-leaflet.js

```diff
@@ -36,12 +36,14 @@ include.module( 'layer-leaflet.layer-esri-feature-leaflet-js', [ 'layer.layer-es
                 // cfg.drawingInfo.renderer.symbol.url = this.resolveUrl( cfg.drawingInfo.renderer.symbol.url )
                 cfg.drawingInfo.renderer.symbol.url = ( new URL( cfg.drawingInfo.renderer.symbol.url, document.location ) ).toString()
         }
-        
+
         var layer = L.esri.featureLayer( cfg )
         
         if ( layers[ 0 ].legendCacheResolve ) {
+            const hideLegend = layers[0].config.hideLegend ? layers[0].config.hideLegend : "";
             layer.legend( function ( err, leg ) {
-                layers[ 0 ].legendCacheResolve( err ? null : leg.layers[ 0 ].legend )
+                layers[0].legendCacheResolve(err ? null : 
+                    leg.layers[0].legend.filter(lg => lg.values.every(v => !hideLegend.includes(v))));
```

Comment:
What are the requirements for the hideLegend config item? Is it a value representing a single string? Space delimited strings? Comma delimited strings?

It might be more intuitive from a configuration point of view for hideLegend to be an array of strings. You could them do a case-insensitive comparison of your legend values against each element of hideLegend instead of hideLegend.includes(v).

Reply:
Good point about case - I'll update for that, and I should be able to work out how to do an array <-> array comparison here.